Sources/SmithyReadWrite/WireResponseErrorClosure.swift

```diff
@@ -0,0 +1,27 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/// Defines a closure that can be used to convert a HTTP response to a Swift `Error`.
+public typealias WireResponseErrorClosure<WireResponse> = (WireResponse) async throws -> Error
+
+/// Defines a closure that can be used to convert a HTTP response and `Reader` for that response to a Swift `Error`.
+public typealias WireResponseErrorBinding<WireResponse: WireDataProviding, Reader: SmithyReader> =
+    (WireResponse, WireResponseDocumentBinding<WireResponse, Reader>) async throws -> Error
```

Comment:
These typealiases replace the former `HTTPResponseErrorClosure ` and `HTTPResponseErrorBinding` with closures that are not dependent on HTTP.

Otherwise, this is just moved in from ClientRuntime.